src/libraries/System.Linq/src/System/Linq/OrderBy.cs

```diff
@@ -108,60 +94,16 @@ public static partial class Enumerable
             new OrderedEnumerable<TSource, TKey>(source, keySelector, comparer, true, null);
 
         public static IOrderedEnumerable<TSource> ThenBy<TSource, TKey>(this IOrderedEnumerable<TSource> source, Func<TSource, TKey> keySelector)
-        {
-            if (source == null)
```

Comment:
Seems you've also deleted the null check for `source` in a couple of places. That should remain.

Reply:
Why? The ctor of `OrderedEnumerable` already checks for `source`.

Reply:
Oh I see it, changing it..